hub/api/dataset_utils.py

```diff
@@ -0,0 +1,39 @@
+from typing import Union
```

Comment:
Can you move this to https://github.com/activeloopai/Hub/tree/release/2.0/hub/util

We have utils file scattered all over.

Reply:
i don't think we should have all utils in one directory, it makes sense to have them hierarchically organized

Reply:
That makes us have a lot of util classes everywhere. Also they make the generated API docs a bit hard to read.